pyogrio/geopandas.py

```diff
@@ -552,6 +565,61 @@ def write_dataframe(
         else:
             crs = geometry.crs.to_wkt(WktVersion.WKT1_GDAL)
 
+    if use_arrow:
+        import pyarrow as pa
+        from pyogrio.raw import write_arrow
+
+        if geometry_column is not None:
+            # Convert to multi type
+            if promote_to_multi:
+                import shapely
+
+                mask_points = geometry_types_all == "Point"
+                mask_linestrings = geometry_types_all == "LineString"
+                mask_polygons = geometry_types_all == "Polygon"
+
+                if mask_points.any():
+                    geometry[mask_points] = shapely.multipoints(
+                        np.atleast_2d(geometry[mask_points]), axis=0
+                    )
+
+                if mask_linestrings.any():
+                    geometry[mask_linestrings] = shapely.multilinestrings(
+                        np.atleast_2d(geometry[mask_linestrings]), axis=0
+                    )
+
+                if mask_polygons.any():
+                    geometry[mask_polygons] = shapely.multipolygons(
+                        np.atleast_2d(geometry[mask_polygons]), axis=0
+                    )
+
+            geometry = to_wkb(geometry.values)
+            df = df.copy(deep=False)
+            # convert to plain DataFrame to avoid warning from geopandas about
+            # writing non-geometries to the geometry column
+            df = pd.DataFrame(df, copy=False)
+            df[geometry_column] = geometry
+        table = pa.Table.from_pandas(df)
```

Comment:
```suggestion
        table = pa.Table.from_pandas(df, preserve_index=False)
```

(per the errors on CI)

Reply:
I would prefer to keep the default behaviour of pyarrow (the CI failures have been fixed, that due to 1) older versions of pandas not using a RangeIndex in certain cases, and 2) filtering rows conditionally for certain parametrization, again causing the index to become not a RangeIndex).

The default behaviour of pyarrow is: "The default of None will store the index as a column, except for RangeIndex which is stored as metadata only" 

Setting it to False would drop the index

Reply:
Ah, but we drop the index for the normal path as well ..

```
In [7]: df = geopandas.GeoDataFrame({"col": [1, 2], "geometry": [shapely.Point(1, 1), shapely.Point(2, 2)]}, index = pd.Index(["a", "b"], name="ids"))

In [8]: df
Out[8]: 
     col                 geometry
ids                              
a      1  POINT (1.00000 1.00000)
b      2  POINT (2.00000 2.00000)

In [9]: pyogrio.write_dataframe(df, "test_index.gpkg")
/home/joris/scipy/repos/pyogrio/pyogrio/raw.py:585: UserWarning: 'crs' was not provided.  The output dataset will not have projection information defined and may not be usable in other systems.
  warnings.warn(

In [10]: pyogrio.read_dataframe("test_index.gpkg")
Out[10]: 
   col                 geometry
0    1  POINT (1.00000 1.00000)
1    2  POINT (2.00000 2.00000)

```